File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int ans = 0;
        int start = 0;
        int end = height.length - 1;
        while (start < end) {
            int minArea = Math.min(height[start], height[end]);
            ans = Math.max(ans, minArea * (end - start));
            if (minArea == height[start])
                start++;
            else
                end--;
        }
        return ans;
    }
}

File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
    let left = 0;
    let right = height.length - 1;
    let res = 0;
    while (left < right) {
        let minValue = Math.min(height[left], height[right]);
        res = Math.max(res, minValue * (right - left));
        if (minValue === height[left]) {
            left++;
        }
        else {
            console.log("right", right);
            right--;
        }
    }
    return res;
};